### PR TITLE
PLAT-129751: ProgressBar, IncrementSlider, Slider Architecture sync

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Issues: 
agate/ProgressBar, Slider and IncrementSlider were missing a tooltip to display the current value

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- added ProgressBarTooltip components
- ProgressBar: added tooltip, highlighted, backgroundProgress prop
- Slider: - added tooltip and backgroundProgress  functionality
- IncrementSlider:  - added tooltip functionality and prop `activateOnFocus` to support slider activation when slider gets focused
- Updates unit tests / ui tests / screenshot tests

### Links
[//]: # (Related issues, references)
PLAT-129751

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: IncrementSlider/IncrementSlider.js [hunk 1]
```diff
@@ -63,6 +64,16 @@ const IncrementSliderBase = kind({
 	name: 'IncrementSlider',
 
 	propTypes: /** @lends agate/IncrementSlider.IncrementSliderBase.prototype */ {
+		/**
+		 * Activates the slider knob when focused so that it may be manipulated via the directional
+		 * input keys.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
Comment (adrian-cocoara-lgp) [inline IncrementSlider/IncrementSlider.js:72]: Is this needed?

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Review by stanca-pop-lgp — APPROVED** [review]
Superb! Great work!

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,11 @@ The following is a curated list of changes in the Enact agate module, newest cha
 
 ### Added
 
+- `agate/IncrementSlider` prop `activateOnFocus` to support slider activation when slider gets focused
+- `agate/IncrementSlider` prop `tooltip` to enable the built-in tooltip
+- `agate/ProgressBar` prop `tooltip`  to enable the built-in tooltip
+- `agate/ProgressBarTooltip` component
```
Comment (0x64) [inline CHANGELOG.md:12]: `agate/SliderTooltip` is also added.

**Review by 0x64 — CHANGES_REQUESTED** [review]

**Comment by 0x64** [comment]
Synced up on `develop` branch to resolve conflicts by #494

File: tests/screenshot/apps/components/ProgressBar.js [hunk 3]
```diff
@@ -4,7 +4,10 @@ import React from 'react';
 const ProgressBarTests = [
 	<ProgressBar />,
 	<ProgressBar orientation="vertical" />,
-	<ProgressBar progress={0.75} size="small" />
+	<ProgressBar progress={0.75} />,
```
[the remark this answers is not in the document]
Comment (daniel-stoian-lgp) [reply tests/screenshot/apps/components/ProgressBar.js:6]: Thanks for the tip. I will add more ss-tests for all 3 components

**Review by 0x64 — APPROVED** [review]
:lgtm:

**Comment by MikyungKim** [comment]
Did you add orientation="radial" props to `ProgressBar`? I'm not seeing the change.

**Comment by daniel-stoian-lgp** [comment]
> Did you add orientation="radial" props to `ProgressBar`? I'm not seeing the change.

I had the intention at the beggining of the implementation, but after some discussions back in December regarding what we should bring to Agate in the architecture sync, and because there was no use case for this orientation in Zeplin, I removed that part from this PR. 
Sorry, I forgot to update the PR description. I will update it now :)

File: ProgressBar/ProgressBar.js [hunk 4]
```diff
@@ -7,14 +7,22 @@
  * @module agate/ProgressBar
```
Comment (MikyungKim) [inline ProgressBar/ProgressBar.js:7]: I'm seeing `backgroundProgress` in the @example but it seems we don't have it.

**Review by MikyungKim — COMMENTED** [review]
@daniel-stoian-lgp , For props `backgroundProgress` and `hightlighted` of ProgressBar, don't we need to sync this?

**Comment by daniel-stoian-lgp** [comment]
> @daniel-stoian-lgp , For props `backgroundProgress` and `hightlighted` of ProgressBar, don't we need to sync this?

hmmm, I see the IncrementSlider already has a backgroundProgress prop which is not used in the sampler. So I think ProgressBar would need this sync.
I'll take care of there two props :)

**Comment by daniel-stoian-lgp** [comment]
Travis fails because of failing unit tests after #522 was merged

**Comment by MikyungKim** [comment]
@daniel-stoian-lgp Please git merge the latest develop on your branch. It will fix the travis.

**Comment by daniel-stoian-lgp** [comment]
> @daniel-stoian-lgp Please git merge the latest develop on your branch. It will fix the travis.

Thanks :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,8 +9,14 @@ The following is a curated list of changes in the Enact agate module, newest cha
 - `agate/ContextualPopup` prop `css` to support customizations
 - `agate/ContextualPopupDecorator` prop `css` to support customizations
 - `agate/Dropdown` prop `width` to support multiple widths
+- `agate/IncrementSlider` prop `activateOnFocus` to support slider activation when slider gets focused
+- `agate/IncrementSlider`, `agate/ProgressBar`, and `agate/Slider` prop `tooltip` to enable the built-in tooltip
 - `agate/Item` to have a `selected` publicClassname
 - `agate/RadioItem` prop `slotBefore`
+- `agate/ProgressBarTooltip` props `highlighted` and `backgroundProgress`
```
Comment (MikyungKim) [inline CHANGELOG.md:16]:
```suggestion
- `agate/ProgressBar` props `highlighted` and `backgroundProgress`
```

File: Slider/Slider.js [hunk 6]
```diff
@@ -72,6 +74,15 @@ const SliderBase = kind({
 		 */
 		active: PropTypes.bool,
 
+		// /**
```
Comment (MikyungKim) [inline Slider/Slider.js:77]: Why these lines are commented out?

Comment (daniel-stoian-lgp) [reply Slider/Slider.js:77]: my mistake

**Review by MikyungKim — APPROVED** [review]
:shipit:

**Pull request merged**